### ISSUE TITLE
terraform validate does not catch name violation for aws_alb_target_group

### ISSUE THREAD

**Issue by gfysaris** [terraform-providers/terraform-provider-aws#6153]
### Terraform Version
Terraform v0.11.8
+ provider.aws v1.40.0

### Affected Resource(s)

* aws_alb_target_group

### Terraform Configuration Files

```hcl
resource "aws_alb_target_group" "alb_tg_app" {
  name     = "tf_alb_tg_app"
  port     = 8080
  protocol = "HTTP"
  vpc_id   = "${aws_vpc.vpc.id}"
}
```

### Expected Behavior

```
aws_alb_target_group.alb_tg_app: Error creating LB Target Group: ValidationError: Target group name 'tf_alb_tg_app' can only contain characters that are alphanumeric characters or hyphens(-)
```
### Actual Behavior

No error returned.

### Steps to Reproduce

1. `terraform validate`

**Comment by tkbky** [comment]
I've made a [PR](https://github.com/terraform-providers/terraform-provider-aws/pull/6167) for this.

**Comment by saravanan30erd** [comment]
@tkbky I think its better to create separate tests for `targetgroup` name and also you should add validation check for `name_prefix` too. You can refer my PR for that. I am happy to close mine.

**Comment by tkbky** [comment]
@saravanan30erd Since yours has covered most of it, let's close mine. 😃

**Comment by bflad** [comment]
Improved plan-time validation has been merged and will release with version 1.42.0 of the AWS provider, sometime in the next few days.
